src/components/Modal/Modal.module.css

```diff
@@ -0,0 +1,140 @@
+@import '../../design-tokens/mixins.css';
+
+/*------------------------------------*\
+    # MODAL
+\*------------------------------------*/
+
+/**
+ * The modal wrapper and overlay which takes up the entire screen.
+ */
+.modal,
+.modal__overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  height: 100vh;
+  width: 100%;
+}
+
+/**
+ * The inverted background of the modal to provide contrast with the actual modal.
+ */
+.modal__overlay {
+  background-color: var(--eds-theme-color-body-background-inverted);
+  opacity: 0.5;
+}
+
+/**
+ * The modal container which positions the modal in the center of the screen.
+ * 1) Set a default z-index on the modal container. This is our smallest z-index token.
+ */
+.modal {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: var(--eds-size-2);
+
+  z-index: var(--eds-z-index-100); /* 1 */
+}
+
+/**
+ * Modal transition animations.
+ */
+.modal__transition--enter {
+  transition: opacity var(--eds-anim-fade-long) var(--eds-anim-ease);
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+.modal__transition--enterFrom {
+  opacity: 0;
+}
+.modal__transition--enterTo {
+  opacity: 1;
+}
+.modal__transition--leave {
+  transition: opacity var(--eds-anim-fade-long) var(--eds-anim-ease);
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+.modal__transition--leaveFrom {
+  opacity: 1;
+}
+.modal__transition--leaveTo {
+  opacity: 0;
+}
+
+/**
+ * The content of the modal, which can wrap header, body, and footer.
+ * 1) This transparent border is for Window High Contrast Mode, which removes all
+    background colors but makes borders 100% opacity black. Without this, the
+    modal would have no clear boundary.
+ *  2) We use flexbox to handle height and width automatically. 
+    The body can scroll when content goes beyond the component,
+    while the footer remains static (if scrollable).
+ */
+.modal__content {
+  position: relative;
+  height: 43.125rem;
```

Comment:
So the thing about the fixed height is, separate from the content scrolling part, if the browser window is too short to show the whole modal, you can't scroll it to see the whole thing, which means the header and footer might be unreachable to a user. For a specific use case, image a user is on a chromebook and has to zoom in to read the text. They can't zoom in and also interact with the modal.

Reply:
My recommendation is this is a max height where the minimum is something like 90% of the viewport.